astropy/coordinates/transformations.py

```diff
@@ -674,6 +674,100 @@ def deco(func):
             return func
         return deco
 
+    def _create_direct_transform(self, fromsys, viasys, tosys, priority=1):
+        """
+        Create and register a direct transform from one frame to another frame
+        via an intermediate frame, using the transforms that already exist in
+        the graph.
+
+        This method is primarily useful for defining loopback transformations
+        (i.e., where ``fromsys`` and ``tosys`` are the same).
+
+        Parameters
+        ----------
+        fromsys : class
+            The coordinate frame class to start from.
+        viasys : class
+            The coordinate frame class to transform through.
+        tosys : class
+            The coordinate frame class to transform into.
+        priority : number
+            The priority if this transform when finding the shortest
+            coordinate transform path - large numbers are lower priorities.
+
+        Notes
+        -----
+        Even though the created transform is a single step in the graph, it
+        will still internally call the constituent transforms.  Thus, there is
+        no performance benefit for using this created transform.
+
+        For Astropy's built-in frames, the most common choice for ``viasys`` is
+        `~astropy.coordinates.ICRS`.
+
+        An error will be raised if a direct transform between ``fromsys`` and
+        ``tosys`` already exists, regardless of whether that existing transform
+        internally uses ``viasys``.
+        """
+        full_path = self.get_transform(fromsys, tosys)
+        if full_path is None:
+            raise ValueError(f"{fromsys.__name__}->{tosys.__name__} is not possible")
+        if len(full_path.transforms) == 1:
+            raise ValueError(f"A direct transform for {fromsys.__name__}->{tosys.__name__} already exists")
+
+        transform_string = f"{fromsys.__name__}->{tosys.__name__} via {viasys.__name__}"
+
+        first_path = self.get_transform(fromsys, viasys)
+        second_path = self.get_transform(viasys, tosys)
+        if first_path is None or second_path is None:
+            raise ValueError(f"{transform_string} is not possible")
+
+        trans_class_set = set([type(t) for t in first_path.transforms] +
+                              [type(t) for t in second_path.transforms])
+
+        # Shared frame attributes are used for instantiating the intermediate frame
+        # For any attributes that are shared by the "from"/"to" frames, the "to" frame takes precedence
```

Comment:
Is it possible to add a note explaining why? I'm sure there's a good reason, but I don't see it immediately, and if there's a good reason it has to be this way, we should make sure the comment is clear so no-one changes it later..

Reply:
In other words, the reasoning needs to be documented in both places.

Reply:
I think so... Isn't the reasoning just that it *has* to be this way for the last step in the chain, and we use the same code for every step?

Reply:
I believe it's simply a choice, with either way being implementable.  I have added a comment explicitly noting that I've implemented the same choice as `CompositeTransform.__call__()`